### ISSUE TITLE
NUnit3TestAdapter 3.17.0 empty output file regression?

### ISSUE THREAD

**Issue by zpqrtbnk** [nunit/nunit3-vs-adapter#780]
Versions: NUnit v3.12.0, NUnit3TestAdapter v3.17.0.
Visual Studio Pro 2019 v16.7.2.
.NET SDKs & runtimes 2.1.804, 3.1.401, 5.0.100-preview.8.

Running tests as:
```
dotnet test my.csproj -c Release --no-restore -f netcoreapp3.1 -v normal -- NUnit.WorkDirectory="(path)" NUnit.TestOutputXml="." NUnit.Labels=Before NUnit.Where="(filter)"
```
With version 3.16.1 of the adapter this produces a result files in `(path)` will complete details of all tests. With version 3.17.0 of the adapter that same file only contains:

```
<?xml version="1.0" encoding="utf-8" standalone="no"?>
<command-line><![CDATA[path\to\bin\Release\netcoreapp3.1\testhost.dll --port 18533 --endpoint 127.0.0.1:018533 --role client --parentprocessid 13248 --telemetryoptedin false]]></command-line>
```

That is, `command-line` as root element without any other content.

Expected: same file as with 3.16.1.

I can repro by switching back and forth the adapter NuGet package between 3.16 and 3.17.

Happy to run more tests or provide more details as needed. Will try to have a look into why this is happening.

**Comment by zpqrtbnk** [comment]
So... at NUnitEngineAdapter.cs:136 there is
```
resultWriter.WriteResultFile(testResults.TopNode, path);
``` 
and at NUnitResults.cs:50 there is
```
TopNode = results.Name == "test-run" ? results.FirstChild : results;
```
which unless I am mistaken makes `TopNode` be the first child of `<test-run>`, i.e. `<command-line>`, thus causing the results file to only contain the `<command-line>` element?

**Comment by sauravwcm** [comment]
Is there a solution or a work around?

**Comment by OsirisTerje** [comment]
@sauravwcm  The fix is done.  Once the PR is through, it should be on the dev feed.  Appreciate confirmations :-)

**Comment by OsirisTerje** [comment]
@zpqrtbnk @sauravwcm Link to the dev package:  https://www.myget.org/feed/nunit/package/nuget/NUnit3TestAdapter/4.0.0-dev-01535

**Comment by sauravwcm** [comment]
Thanks @OsirisTerje The solution implemented in the dev package indeed works. Any info on when NUnit3TestAdapter_4.0.0 will be officially released?

**Comment by zpqrtbnk** [comment]
Cool Thanks!!

**Comment by sauravwcm** [comment]
Has anyone of you observed that "run selected test" feature sometimes fails with  NUnit3TestAdapter 4.0.0-dev-01535 on visual studio 2017

**Comment by OsirisTerje** [comment]
@sauravwcm No really, but what I have observed is that when I add a new test or change the test name (method or class), it needs to be run twice, once to make it reappear in the test explorer, then again for actually running it.  But, not that it sometimes fails or not.  Sounds like something flaky.   If you can drill down that a bit, slim it, and perhaps come up with a  repro ?

**Comment by OsirisTerje** [comment]
@ngyukman Might be able to get one out during easter.  Can you confirm the beta is working as it should for you ?

**Comment by ngyukman** [comment]
@OsirisTerje yes, so far I don't see much issues in my execution

**Comment by Deadvi5** [comment]
NUnit3TestAdapter version 4.3.0 uses with Stryker.NET still have this issue;
Rollback to version 3.17 fixed the issue

**Comment by OsirisTerje** [comment]
Thanks.  See issues #1027 and #1028.  There is an alpha there with a fix, and a 4.3.1 will be out shortly
